synapse/handlers/room_summary.py

```diff
@@ -713,6 +716,8 @@ async def _build_room_entry(self, room_id: str, for_federation: bool) -> JsonDic
             ),
             "guest_can_join": stats["guest_access"] == "can_join",
             "room_type": create_event.content.get(EventContentFields.ROOM_TYPE),
+            "im.nheko.summary.version": stats["version"],
```

Comment:
In some situations this could potentially fix the hack on L656-657 about assuming `knock` is valid. Would probably not be available often enough though.

We should guard these with the unstable config flag, probably?

Reply:
I think we should wait for this to be stable first and available for a while and then we can use that to fix the hack, since we need the remote server to use it. After a few months it is probably safe to just assume a room is version 1, if no version is returned.

I think an unstable flag would probably make more sense for the whole feature instead of just some namespaced fields. I was hoping the MSC just FCPs and passes in the next few weeks, since it is quite small and then we can just stabilize it (so I don't need to figure out how to add unstable flags :D). But I'll defer that to your judgement. If you want a flag, I can add it. I am just lazy .-.

Reply:
> I think an unstable flag would probably make more sense for the whole feature instead of just some namespaced fields. I was hoping the MSC just FCPs and passes in the next few weeks, since it is quite small and then we can just stabilize it (so I don't need to figure out how to add unstable flags :D). But I'll defer that to your judgement. If you want a flag, I can add it. I am just lazy .-.

There already is one for the whole feature -- I'm suggesting to re-use it and guard these fields with it.

Reply:
Okay, I think I figured it out. I cache the flag in the Handler, as it doesn't cache the hs yet and I think just caching the flag makes more sense? No idea :D